pagefaultgames/pokerogue
# [Move] Fix multi-hit moves activating type-immunity abilities multiple times

## What are the changes?
Fix multi-hit moves (i.e. Bullet Seed, Water Shuriken) triggering type-immunity abilities more than once.

## Why am I doing these changes?
Fix #2123.

## What did change?
Refactor `MultiHitAttr` to check the type effectiveness of the attack before assigning the hit count. If the defender is immune to the attack, the attack only gets 1 hit. This check includes type immunity abilities, so the defender's type immunity ability only triggers once.

### Screenshots/Videos



https://github.com/pagefaultgames/pokerogue/assets/6374275/b2ec02f9-42e9-4bc5-90ac-60279d95cf73


## How to test the changes?
The unit tests should handle it. For manual testing, consider the following overrides:

```ts
export const MOVESET_OVERRIDE: Array<Moves> = [Moves.WATER_SHURIKEN];
export const OPP_ABILITY_OVERRIDE: Abilities = Abilities.WATER_ABSORB;
```

## Checklist
